src/tests/ftest/container/snapshot_aggregation.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+import time
+
+from ior_test_base import IorTestBase
+from daos_utils import DaosCommand
+
+
+class SnapshotAggregation(IorTestBase):
+    # pylint: disable=too-many-ancestors
+    """Defines snapshot aggregation test cases.
+
+    Test Class Description:
+          Verify snapshot aggregation.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a SnapshotAggregation object."""
+        super(SnapshotAggregation, self).__init__(*args, **kwargs)
+        self.dmg = None
+        self.free_space = {"scm": [], "nvme": []}
+
+    def update_free_space(self):
+        """Append the free space list with the current pool capacities."""
+        data = self.dmg.pool_query(self.pool.uuid)
+        self.pool.get_info()
+
+        for index, name in enumerate(("scm", "nvme")):
+            self.free_space[name].append({
+                "dmg": data[name]["free"],
+                "api": int(self.pool.info.pi_space.ps_space.s_free[index])
```

Comment:
We have dmg and api keys, but you use api everywhere. Is there a reason? Are they interchangeable?

Reply:
Originally I was planning on updating the `TestPool.get_info()` to support either the `dmg` or `api` control method and return the same data structure regardless of which control method was used, but the output from the dmg command did not map well and it became a larger-than-the-original-PR issue to resolve for all the tests that make use of the `TestPool.get_info()` method.  So for now I'm calling both methods, using the dmg output to display the free space (human readable) and the API output to do the comparison (values in the same units).

Reply:
I see. dmg pool query returns a lot of stuff, so I can imagine it's hard to map the output with the API. Thanks.